### PR TITLE
PPS clean code for bad pot

### PR DESCRIPTION
**Pull request by fabferro**
#### PR description:

This PR is a follow-up of PR #38553 implemented the requested code cleaning. No changes expected. 

#### PR validation:

Run3 data processing.

No backport needed.

@tvami @grzanka

### CODE REVIEW

**Comment by clacaputo** [comment]
+reconstruction

- follow-up of PR https://github.com/cms-sw/cmssw/pull/38553
- no reco changes

File: RecoPPS/Local/plugins/CTPPSPixelLocalTrackProducer.cc [hunk 1]
```diff
@@ -65,31 +65,31 @@ class CTPPSPixelLocalTrackProducer : public edm::stream::EDProducer<> {
 
   edm::InputTag inputTag_;
```
Comment (perrotta) [inline RecoPPS/Local/plugins/CTPPSPixelLocalTrackProducer.cc:66]: `inputTag_` never used

Comment (perrotta) [inline RecoPPS/Local/plugins/CTPPSPixelLocalTrackProducer.cc:66]: `inputTag_` never used

File: RecoPPS/Local/plugins/CTPPSPixelLocalTrackProducer.cc [hunk 2]
```diff
@@ -65,31 +65,31 @@ class CTPPSPixelLocalTrackProducer : public edm::stream::EDProducer<> {
...
   edm::EDGetTokenT<edm::DetSetVector<CTPPSPixelRecHit>> tokenCTPPSPixelRecHit_;
```
Comment (perrotta) [inline RecoPPS/Local/plugins/CTPPSPixelLocalTrackProducer.cc:67]: why not to make `const` all other classs members initialized in the ctor initialization list (and never modified afterwards)?

File: RecoPPS/Local/plugins/CTPPSPixelLocalTrackProducer.cc [hunk 3]
```diff
@@ -65,31 +65,31 @@ class CTPPSPixelLocalTrackProducer : public edm::stream::EDProducer<> {
...
-  edm::ESGetToken<CTPPSGeometry, VeryForwardRealGeometryRecord> tokenCTPPSGeometry_;
+  const edm::ESGetToken<CTPPSGeometry, VeryForwardRealGeometryRecord> tokenCTPPSGeometry_;
   edm::ESWatcher<VeryForwardRealGeometryRecord> geometryWatcher_;
 
-  edm::ESGetToken<CTPPSPixelAnalysisMask, CTPPSPixelAnalysisMaskRcd> tokenCTPPSPixelAnalysisMask_;
+  const edm::ESGetToken<CTPPSPixelAnalysisMask, CTPPSPixelAnalysisMaskRcd> tokenCTPPSPixelAnalysisMask_;
 
   uint32_t numberOfPlanesPerPot_;
   std::vector<uint32_t> listOfAllPlanes_;
```
Comment (perrotta) [inline RecoPPS/Local/plugins/CTPPSPixelLocalTrackProducer.cc:74]: This is only used locally in the constructor: no need to make it a class member.
Moreover: why to `reserve(6)` for it? Are you hardcoding somehow the number of elements that should be passed with the `numberOfPlanesPerPot_` parameter instead? In any case, if this variable becomes local in the constructor, there is no need to reserve its size

**Comment by qliphy** [comment]
@fabferro Please have a look at above comments.  Moreover, this PR should be backport to 12_4_X

**Comment by fabferro** [comment]
@perrotta : comments implemented. Backport PR opened #38797

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
@cms-sw/reconstruction-l2 could you please check and (re)sign, if you agree with the latest changes?

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge
- @clacaputo already signed this once, and now also the pretty identical 12_4_X backport, see https://github.com/cms-sw/cmssw/pull/38797#issuecomment-1192714691
- I assume that the intention was to also re-sign this one in the master

**Comment by jfernan2** [comment]
+1 (for the reords)

**Pull request merged**